core/arch/arm/kernel/user_ta.c

```diff
@@ -313,6 +313,14 @@ static TEE_Result ta_load(const TEE_UUID *uuid,
 		goto error_return;
 	}
 
+	if (ta_head->flags & TA_FLAG_MULTI_SESSION &&
```

Comment:
I don't fully agree with this approach. If it's a problem that those flags may be set then they shouldn't have been set from the beginning and we should just refuse to load the TA.